beacon-light-client/plonky2/crates/circuits/src/deposit_accumulator_balance_aggregator_diva/first_level.rs

```diff
@@ -0,0 +1,505 @@
+use crate::{
+    common_targets::{DepositDivaTarget, Sha256MerkleBranchTarget},
+    deposits_accumulator_balance_aggregator::common_targets::{
+        AccumulatedDataTargetDiva, ValidatorStatusStatsTarget,
+    },
+    serializers::{
+        biguint_to_str, parse_biguint, serde_bool_array_to_hex_string,
+        serde_bool_array_to_hex_string_nested,
+    },
+    utils::circuit::{
+        assert_bool_arrays_are_equal, get_balance_from_leaf,
+        hashing::{
+            merkle::{
+                poseidon::{hash_validator_poseidon, validate_merkle_proof_poseidon},
+                sha256::validate_merkle_proof_sha256,
+            },
+            poseidon::poseidon,
+        },
+        select_biguint,
+        validator_status::{get_validator_relevance, get_validator_status},
+    },
+};
+use circuit::{circuit_builder_extensions::CircuitBuilderExtensions, Circuit, ToTargets};
+use circuit_derive::{CircuitTarget, SerdeCircuitTarget};
+use num::{BigUint, FromPrimitive};
+use plonky2::{
+    field::goldilocks_field::GoldilocksField,
+    hash::hash_types::HashOutTarget,
+    iop::target::BoolTarget,
+    plonk::{
+        circuit_builder::CircuitBuilder, circuit_data::CircuitConfig,
+        config::PoseidonGoldilocksConfig,
+    },
+};
+use plonky2_crypto::biguint::{BigUintTarget, CircuitBuilderBiguint};
+
+use crate::common_targets::{PoseidonMerkleBranchTarget, Sha256Target, ValidatorTarget};
+
+pub struct DepositAccumulatorBalanceAggregatorDivaFirstLevel {}
+
+#[derive(CircuitTarget, SerdeCircuitTarget)]
+#[serde(rename_all = "camelCase")]
+pub struct DepositAccumulatorBalanceAggregatorDivaFirstLevelTarget {
+    #[target(in)]
+    pub validator: ValidatorTarget,
+
+    #[target(in)]
+    pub commitment_mapper_branch: PoseidonMerkleBranchTarget<40>,
```

Comment:
rename this to `validators_commitment_mapper_branch` as we have multiple commitment mappers already